starclass/BaseClassifier.py

```diff
@@ -255,12 +260,44 @@ def test(self, tset, save=None):
 		# Convert labels to ndarray:
 		# FIXME: Only comparing to the first label
 		y_pred = np.array(y_pred)
+		y_prob = np.array(y_prob)
 		labels_test = self.parse_labels(tset.labels_test())
 
 		# Compare to known labels:
 		acc = accuracy_score(labels_test, y_pred)
 		logger.info('Accuracy: %.2f%%', acc*100)
 
+		f1_macro = f1_score(labels_test, y_pred, average='macro')
+		f1_weighted = f1_score(labels_test, y_pred, average='weighted')
+		logger.info('Macro F1 score: %.2f%%', f1_macro*100)
+		logger.info('Weighted F1 score: %.2f%%', f1_weighted*100)
+
+		# Prepare input for ROC/AUC
+		fpr, tpr, roc_auc, threshold_index, best_thresholds = self.compute_ROC(labels_test, y_prob, all_classes)
+
+		# Create plot of ROC curves:
+		fig = plt.figure(figsize=(12,12))
+		plotROC(fpr, tpr, roc_auc, threshold_index, all_classes)
+		plt.title('ROC - ' + self.classifier_key + ' - ' + tset.key + ' - ' + tset.level)
+		fig.savefig(os.path.join(self.data_dir, 'ROC_Curve_' + tset.key + '_' + tset.level + '_' + self.classifier_key + '.png'), bbox_inches='tight')
+		plt.close(fig)
+
+		# Save new classification thresholds and make final predictions based on those
+		if self.classifier_key == 'meta':
+			self.classification_thresholds = best_thresholds
+			# TODO: move this part to the meta-classifier that will use the saved thresholds to make a final list with predictions (only during runtime, not train/test)
+			new_pred = np.zeros(len(y_pred), dtype='<U15')
+			for i in range(len(y_prob)):
+				idx, max_prob = np.argmax(y_prob[i,:]), np.max(y_prob[i,:])
+				if max_prob > best_thresholds[idx]:
+					new_pred[i] = all_classes[idx]
+				else:
+					new_pred[i] = 'UNKOWN'
+
+		# Classification report for meta
```

Comment:
How about we instead add some kind of function hook, where each individual classifier can execute some after the training completes?